client/blocks/follow-button/index.jsx

```diff
@@ -26,6 +30,10 @@ class FollowButtonContainer extends Component {
 	};
 
 	handleFollowToggle = ( following ) => {
+		if ( ! this.props.isLoggedIn ) {
+			const { pathname } = getUrlParts( window.location.href );
+			return navigate( createAccountUrl( { redirectTo: pathname } ) );
+		}
```

Comment:
I wanted to point out that this isn't ideal. This component likely needs little to no data when the user is logged out, as compared to when the user is logged in.

I'd recommend exploring separate components for logged-in users and logged-out users, where the logged-out one will end up being way simpler and smaller. This helps for better separation of concerns, on top of the performance and bundle size aspects.

That applies to the rest of the areas where we're suggesting support for logged-out users.

Reply:
hmm I understand the point that in the logged out state, we're loading a lot of code that can never be hit which makes the bundle bigger than it needs to be. 

When you talk about data, do you mean data being pulled in from `state` which is already loaded by the app, or new data by requested from the server? I think the problem would be if we're loading unnecessary data from the server, whereas passing existing data within the app doesn't seem like a problem.

I tried to make sure that no unnecessary requests are made (they would fail anyway if they require auth).

In investigating I actually just noticed that there are still calls being made to `/me/settings` and `/me/preferences`, so I'll track those down too 👀

If we want to reduce the bundle size with separate components, it would require a fair bit of duplication of code. So I think it's better not to unless we know that there are meaningful performance problems that we can improve by reducing the bundle size.

Reply:
> In investigating I actually just noticed that there are still calls being made to /me/settings and /me/preferences, so I'll track those down too 👀

NVM it looks like those requests were coming from the environment badge injected on localhost

Reply:
I'm basically talking about anything that the component would require, but won't use for a logged-out user. That could include:

* Any component that we require, but don't render
* Any action creator that we include, but will not be using
* Any styles that we load, but we won't use
* Any component business logic that we don't use.

And it's not only the performance / bundle size aspect; it's also the developer experience aspect - why do I load all these things when I won't be using them? What will happen if we add a few more conditions like that? How will someone new to the codebase understand what's going on? What's the chance that these not-so-good practices will be spread throughout the codebase as someone is using that example as a base for their new work?

Reply:
You can see an example of this problem when we [pulled the blogging prompts card into reader](https://github.com/Automattic/wp-calypso/pull/72923#issuecomment-1414148141). The reader bundle size increased by over 4%, just from adding a fairly simple card component and a little bit of code to determine where to render it.

The card component itself doesn't seem that large, so I'm guessing it's because of all the associated state code required by that component and its dependencies.

Given a primary goal of making these tags pages is SEO related, and bundle size/load speed is an important part of page ranking these days, I suspect it's very much worth investigating how we can reduce the bundle size for logged out pages.

Maybe we can continue with this PR behind a feature flag, and then go through the components one at a time to see where we can reduce the size?

Reply:
ok, I agree, I think it makes sense to start with this as is, and then look at reducing the bundle size in follow up PRs.